vault/resource_transit_cache_config.go

```diff
@@ -0,0 +1,86 @@
+package vault
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/vault/api"
+)
+
+func transitSecretBackendCacheConfig() *schema.Resource {
+	return &schema.Resource{
+		Create: transitSecretBackendCacheConfigUpdate,
+		Update: transitSecretBackendCacheConfigUpdate,
+		Read:   transitSecretBackendCacheConfigRead,
+		Delete: transitSecretBackendCacheConfigDelete,
+
+		Schema: map[string]*schema.Schema{
+			"backend": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The Transit secret backend the resource belongs to.",
+				ForceNew:    true,
+			},
+			"size": {
+				Type:        schema.TypeInt,
+				Description: "Number of cache entries. A size of 0 mean unlimited.",
+				Required:    true,
+			},
+		},
+	}
+}
+
+func transitSecretBackendCacheConfigUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*api.Client)
+	size := d.Get("size").(int)
+
+	backend := d.Get("backend").(string) + "/cache-config"
+
+	log.Printf("[DEBUG] Setting transit cache size to: %d", size)
+	d.SetId(backend)
```

Comment:
We probably don't want to write the ID of the backend until it completes successfully, right? I don't see it explicitly stated in [Terraform Plugin Best Practices][1] but I believe you want to call `SetId` as late as possible, when _the thing_ has actually been created. Otherwise there's the risk that the State gets recorded as having a thing, but it doesn't actually exist. 

[1]: https://www.terraform.io/docs/extend/best-practices/index.html

Reply:
Moved it to the end